pkg/proc/gdbserial/gdbserver.go

```diff
@@ -269,17 +278,11 @@ func (p *Process) Connect(conn net.Conn, path string, pid int) error {
 	if addr, err := p.conn.allocMemory(256); err == nil {
 		if _, err := p.conn.writeMemory(uintptr(addr), p.loadGInstr()); err == nil {
 			p.loadGInstrAddr = addr
+		} else {
```

Comment:
I'd revert this. We have other ways to load g if this fails and this wouldn't have detected your bug anyway (the problem wasn't writing the instruction, it was in executing it later on).